aws_lambda_powertools/metrics/provider/datadog/datadog.py

```diff
@@ -0,0 +1,291 @@
+        """Add cold start metric and function_name dimension
+
+        Parameters
+        ----------
+        context : Any
+            Lambda context
+        """
+        logger.debug("Adding cold start metric and function_name tagging")
+        self.add_metric(name="ColdStart", value=1, function_name=context.function_name)
+
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Dict, Any], Any] | Optional[Callable[[Dict, Any, Optional[Dict]], Any]] = None,
+        capture_cold_start_metric: bool = False,
+        raise_on_empty_metrics: bool = False,
+        **kwargs,
+    ):
+        """Decorator to serialize and publish metrics at the end of a function execution.
+
+        Be aware that the log_metrics **does call* the decorated function (e.g. lambda_handler).
+
+        Example
+        -------
+        **Lambda function using tracer and metrics decorators**
+
+            from aws_lambda_powertools import DatadogMetrics, Tracer
+
+            metrics = DatadogMetrics(namespace="powertools")
+            tracer = Tracer(service="payment")
+
+            @tracer.capture_lambda_handler
+            @metrics.log_metrics
+            def handler(event, context):
+                    ...
+
+        Parameters
+        ----------
+        lambda_handler : Callable[[Any, Any], Any], optional
+            lambda function handler, by default None
+        capture_cold_start_metric : bool, optional
+            captures cold start metric, by default False
+        raise_on_empty_metrics : bool, optional
+            raise exception if no metrics are emitted, by default False
+        **kwargs
+
+        Raises
+        ------
+        e
+            Propagate error received
+        """
+
+        default_tags = kwargs.get("default_tags")
+
+        if default_tags:
+            self.set_default_tags(**default_tags)
+
+        return super().log_metrics(
+            lambda_handler=lambda_handler,
+            capture_cold_start_metric=capture_cold_start_metric,
+            raise_on_empty_metrics=raise_on_empty_metrics,
+            **kwargs,
+        )
+
+    def set_default_tags(self, **kwargs) -> None:
+        """Persist tags across Lambda invocations
+
+        Parameters
+        ----------
+        tags : **kwargs
+            tags as key=value
+
+        Example
+        -------
+        **Sets some default dimensions that will always be present across metrics and invocations**
+
+            from aws_lambda_powertools import Metrics
+
+            metrics = Metrics(namespace="ServerlessAirline", service="payment")
+            metrics.set_default_tags(environment="demo", another="one")
+
+            @metrics.log_metrics()
+            def lambda_handler():
+                return True
+        """
+        for tag_key, tag_value in kwargs.items():
+            tag = f"{tag_key}:{tag_value}"
+            if tag not in self.default_tags:
+                self.default_tags.append(tag)
```

Comment:
after you use Dict for default tags, this will simply become a dict update operation. Then you'll use a list comprehension to unpack those tags into Datadog's expected format.

If Datadog later changes or improves its format, you only change the serialization piece. Another benefit is that a Dict is a hash table, so you can't have the same dict key name, simplifying maintenance.